### PR TITLE
Extract CreateMemoryTable, DropTable, CreateExternalTable in LogicalPlan as independent struct

### LINKED ISSUE #1306: Extract `CreateMemoryTable`, `DropTable`, `CreateExternalTable` in LogicalPlan as independent struct
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
There are some discussions in #1228

**Describe the solution you'd like**
like #1290

**Describe alternatives you've considered**
The task is mechanical, however, it is ideal for newcomers to familiarize themselves with datafusion's build, test, and PR processes.

**Additional context**
If you want to do it, please ping me and assign the task to you, avoiding collisions with other people.

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Closes #1306 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by liukun4515** [comment]
@xudong963  PTAL

**Comment by xudong963** [comment]
Thanks @liukun4515, I'll take a look tomorrow!

File: datafusion/src/logical_plan/mod.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ mod display;
 mod expr;
 mod extension;
 mod operators;
-pub mod plan;
+mod plan;
```
Comment (Jimexist) [inline datafusion/src/logical_plan/mod.rs:30]: I think this is not backward compatible

**Comment by alamb** [comment]
This PR now also has several logical conflicts with master :(

File: datafusion/src/logical_plan/mod.rs (hunk 1 above)
Comment (alamb) [reply datafusion/src/logical_plan/mod.rs:30]: I agree with @Jimexist 's implicit suggestion that we should leave the code as `pub mod plan;`

**Review by alamb — APPROVED** [review]
Thanks @liukun4515  -- can you please address @Jimexist 's comment and resolve the merge conflicts and I think this one will be ready to merge

**Comment by xudong963** [comment]
LGTM!

File: datafusion/src/logical_plan/mod.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/src/logical_plan/mod.rs]: @Jimexist 
@alamb 
This comment has been addressed

Comment (liukun4515) [reply datafusion/src/logical_plan/mod.rs]: @Jimexist @alamb 
I have doubts about this backward compatible.
Before this commit #1290, this is `mod plan`, and this commit is not in the release 6.0.0 datafusion.
The reason i did this is to keep code style with original usage:
```
pub use plan::{
    JoinConstraint, JoinType, LogicalPlan, Partitioning, PlanType, PlanVisitor,
};
```
and make some struct public we wanted.

**Review by Jimexist — APPROVED** [review]
LGTM thanks

**Comment by liukun4515** [comment]
@alamb  please merge this pr

**Comment by alamb** [comment]
Thanks @liukun4515 !

**Pull request merged**